src/sentry/tasks/invalid_group_status_detection.py

```diff
@@ -0,0 +1,87 @@
+import random
+from collections import defaultdict
+from datetime import datetime, timedelta, timezone
+from logging import getLogger
+
+from django.db import connection
+
+from sentry.models.group import Group, GroupStatus
+from sentry.models.grouphistory import GroupHistoryStatus
+from sentry.silo import SiloMode
+from sentry.tasks.auto_ongoing_issues import TRANSITION_AFTER_DAYS
+from sentry.tasks.base import instrumented_task
+from sentry.types.group import SUBSTATUS_TO_STR, UNRESOLVED_SUBSTATUS_CHOICES, GroupSubStatus
+from sentry.utils.query import RangeQuerySetWrapper
+
+logger = getLogger(__name__)
+
+SAMPLE_SIZE = 500
+SEVEN_DAYS_AGO = datetime.now(tz=timezone.utc) - timedelta(days=TRANSITION_AFTER_DAYS)
+QUERY = """SELECT group_id, MAX(date_added)
+    FROM sentry_grouphistory
+    WHERE group_id IN ({group_ids}) and status = {status}
+    GROUP BY group_id
+"""
+
+
+@instrumented_task(
+    name="sentry.tasks.issues.invalid_group_status_detection",
+    queue="invalid_group_status_detection",
+    time_limit=5 * 60,  # 5 minutes
+    max_retries=3,
+    default_retry_delay=60,
+    silo_mode=SiloMode.REGION,
+)
+def detect_invalid_group_status() -> None:
+    group_ids = list(Group.objects.values_list("id", flat=True))
+    random_group_ids = random.sample(group_ids, SAMPLE_SIZE)
```

Comment:
@snigdhas This is going to generate a list of _every_ single issue from postgres in memory. This is not going to work. We also need to have a better strategy of just picking 500 random issues from all time. I fear that is not enough to uncover any issues.

Reply:
Just in terms of how to efficiently fetch rows randomly

```
select *
from sentry_groupedmessage 
TABLESAMPLE SYSTEM(0.003)
where last_seen > now() - interval '7 days';
```

This is still reasonably slow, a few seconds to run, but if it's infrequent it's not a huge deal. We should also probably run it off of a read replica.

We probably want to limit the selection of rows to some recent time window for this to be effective.